programs/local/LocalServer.cpp

```diff
@@ -414,7 +414,13 @@ try
 {
     UseSSL use_ssl;
     ThreadStatus thread_status;
-    signal(SIGINT, signalHandler);
+
+    struct sigaction new_act;
+    struct sigaction old_act;
+    new_act.sa_handler = interruptSignalHandler;
+    sigemptyset(&new_act.sa_mask);
+    new_act.sa_flags = 0;
+    sigaction(SIGINT, &new_act, &old_act);
```

Comment:
Can these functions fail?

Reply:
Theoretically yes.

Reply:
Should we check if function failed and do something in this case?